### DISCUSSION TITLE
Yolov8 and Yolov5u TensorRT engines not deserializing

### DISCUSSION THREAD

**Discussion by GregBello** [ultralytics/ultralytics#4765] [Q&A]
Has anyone else encountered an issue in the last few weeks where Yolov8 and Yolov5u TensorRT engines do not deserialize? I am trying to test yolov8 in a c++ app and I am using cuda 11.8, cudnn 8.8 and tensorrt 8.5.2.2. The v8 and v5u models when serialized using the newer yolo export... cli command will not deserialize in my own app. However, using all the same dependencies, python export.py ... cli command works when running out of the yolov5 repo and creating the old anchored version of the engines. Let me know if anyone has encountered this problem or can see that I am doing something obviously wrong. 

I am aware that the deserialization fail is usually due to a mismatch in TensorRT versions and maybe other dependencies like cudnn etc. I am certain the version I am linking against is the same I am using when deserializing. Thanks in advance! Let me know if there are follow up questions to gain further clarification. 

The following are the dependencies in my anaconda env: 

absl-py                 1.4.0
asttokens               2.2.1
backcall                0.2.0
brotlipy                0.7.0
cachetools              5.3.1
certifi                 2023.7.22
cffi                    1.15.1
chardet                 5.2.0
charset-normalizer      2.0.4
colorama                0.4.6
coloredlogs             15.0.1
contourpy               1.1.0
cryptography            41.0.2
cycler                  0.11.0
decorator               5.1.1
executing               1.2.0
filelock                3.12.2
flatbuffers             23.5.26
fonttools               4.42.0
gitdb                   4.0.10
GitPython               3.1.32
google-auth             2.22.0
google-auth-oauthlib    1.0.0
grpcio                  1.57.0
humanfriendly           10.0
idna                    3.4
importlib-metadata      6.8.0
importlib-resources     6.0.1
ipython                 8.12.2
jedi                    0.19.0
Jinja2                  3.1.2
kiwisolver              1.4.4
Markdown                3.4.4
markdown-it-py          3.0.0
MarkupSafe              2.1.1
matplotlib              3.7.2
matplotlib-inline       0.1.6
mdurl                   0.1.2
mkl-fft                 1.3.6
mkl-random              1.2.2
mkl-service             2.4.0
mpmath                  1.3.0
networkx                3.1
numpy                   1.23.1
nvidia-pyindex          1.0.9
oauthlib                3.2.2
onnx                    1.14.0
onnxruntime-gpu         1.15.1
onnxsim                 0.4.33
opencv-python           4.8.0.76
packaging               23.1
pandas                  2.0.3
parso                   0.8.3
pickleshare             0.7.5
Pillow                  10.0.0
pip                     23.2.1
prompt-toolkit          3.0.39
protobuf                4.24.0
psutil                  5.9.5
pure-eval               0.2.2
py-cpuinfo              9.0.0
pyasn1                  0.5.0
pyasn1-modules          0.3.0
pycparser               2.21
Pygments                2.16.1
pyOpenSSL               23.2.0
pyparsing               3.0.9
pyreadline3             3.4.1
PySocks                 1.7.1
python-dateutil         2.8.2
pytz                    2023.3
PyYAML                  6.0.1
requests                2.31.0
requests-oauthlib       1.3.1
rich                    13.5.2
rsa                     4.9
scipy                   1.10.1
seaborn                 0.12.2
setuptools              68.0.0
six                     1.16.0
smmap                   5.0.0
stack-data              0.6.2
style                   1.1.0
sympy                   1.11.1
tensorboard             2.14.0
tensorboard-data-server 0.7.1
tensorrt                8.5.2.2
thop                    0.1.1.post2209072238
torch                   2.0.1+cu118
torchaudio              2.0.2+cu118
torchvision             0.15.2+cu118
tqdm                    4.66.1
traitlets               5.9.0
typing_extensions       4.7.1
tzdata                  2023.3
ultralytics             8.0.162
update                  0.0.1
urllib3                 1.26.16
wcwidth                 0.2.6
Werkzeug                2.3.7
wheel                   0.38.4
win-inet-pton           1.1.0
zipp                    3.16.2

**Comment by glenn-jocher** [comment]
@GregBello hello! It seems you're experiencing issues with deserializing YOLOv8 and YOLOv5u TensorRT engines in your C++ application. You've mentioned that you're using the same versions of CUDA, cuDNN, and TensorRT for both serialization and deserialization, which is good practice to avoid compatibility issues.

Given that you're certain about the version consistency, it might be worth checking a few other things:

1. Ensure that the environment used for serialization is identical to the one used for deserialization, including any potential minor version differences in dependencies.
2. Verify that the serialized engine files haven't been corrupted or altered between the serialization and deserialization steps.
3. Check for any architecture-specific optimizations during serialization that might not be compatible with the system you're deserializing on.
4. Review the C++ application's error logs or output for any specific error messages that could provide more insight into the deserialization failure.

If the issue persists, consider isolating the problem by attempting to deserialize a simple, freshly serialized engine on the same system using a minimal setup. This could help determine if the problem is with the specific models or the setup itself.

For further guidance, you might want to refer to the Export mode documentation on our Docs, which provides information on exporting models for deployment, including potential issues and troubleshooting steps.

If you continue to face difficulties, please provide any error messages or logs you're encountering, and we'll do our best to assist you further. Remember, the community and the Ultralytics team are here to help! 😊
